src/core/init/media_source_content_initializer.ts

```diff
@@ -445,13 +445,14 @@ export default class MediaSourceContentInitializer extends ContentInitializer {
       }
     }, { clearSignal: cancelSignal, emitCurrentValue: true });
 
-    const streamObserver = createStreamPlaybackObserver(manifest,
-                                                        playbackObserver,
+    const streamObserver = createStreamPlaybackObserver(playbackObserver,
```

Comment:
NOTE: I also changed a little that function's parameters which mixes objects and values without a very clear reason.